input/pagecontent/StructureDefinition-ViewDefinition-notes.md

```diff
@@ -28,15 +28,24 @@ All view runners must implement these FHIRPath capabilities:
 All view runners must implement these functions that do not exist in the
 FHIRPath specification but are necessary in the context of defining views:
 
-##### getId()
+##### getRowKey([resource: ResourceTypeCode]) : String
 
-Returns the resource ID from
-a [reference](https://hl7.org/fhir/references.html#Reference) element. This
-function has no parameters. Note that implementations of `getId` may differ
-based on the structure of the underlying data being queried. For example,
-potential approaches could include dynamically extracting the last segment from
-the reference URL, looking up the ID from an annotation stored in the annotation
-layer, or calculating a hash of the full reference URL.
+Returns a string that can be used as a database key. It can be invoked as a function on 
```

Comment:
Must it be a string? Some databases may use integers or other types for these references. Since the values will only be used (?) With a quality checks inside of a join constraint, do we need to specify a type?

Reply:
yes, it must be a string, but it could be a string that is an integer in some costs, but if you're Nikolai, you'll be using resource.id directly, so it will only accidentally be an integer